core/src/uix/core.cljs

```diff
@@ -213,3 +213,53 @@
   (let [lazy-component (react/lazy #(.then (f) (fn [component] #js {:default component})))]
     (set! (.-uix-component? lazy-component) true)
     lazy-component))
+
+(defn react-component-ctor []
+  (fn [^js/React.Component this _]
+    (set! (.-state this) #js {:argv nil})
+    (specify! (.-state this)
+              IDeref
+              (-deref [_o]
+                      (.. this -state -argv))
+              IReset
+              (-reset! [_o new-value]
+                       (.setState this #js {:argv new-value})
+                       new-value)
+              ISwap
+              (-swap!
+               ([o f]
+                (-reset! o (f (-deref o))))
+               ([o f a]
+                (-reset! o (f (-deref o) a)))
+               ([o f a b]
+                (-reset! o (f (-deref o) a b)))
+               ([o f a b xs]
+                (-reset! o (apply f (-deref o) a b xs)))))))
+
+(defn create-error-boundary
+  "Creates React's error boundary component
+
+  display-name       — the name of the component to be displayed in stack trace
+  derive-error-state — maps error object to component's state that is used in render-fn
+  did-catch          — 2 arg function for side-effects, logging etc.
+  receives the exception and additional component info as args
+  render-fn          — takes state value returned from error->state and a vector
+  of arguments passed into error boundary"
+  [{:keys [display-name derive-error-state did-catch]
+    :or   {display-name (str (gensym "uix.error-boundary"))}}
+   render-fn]
+  (let [constructor  (react-component-ctor)
+        derive-state (fn [error]
+                       #js {:argv (derive-error-state error)})
+        render       (fn []
+                       (this-as this
+                                (let [args  (.-argv (.-props this))
+                                      state (.-state this)]
+                                  (render-fn state args))))
+        class        (create-class {:constructor              constructor
+                                    :displayName              display-name
+                                    :getDerivedStateFromError derive-state
+                                    :componentDidCatch        did-catch
+                                    :render                   render})]
+    (fn [& args]
+      (react/createElement class #js {:argv args}))))
```

Comment:
I _think_ if you add `(set! (.-uix-component? class) true)` and just return the class itself here instead of a function, then `$` will pick it up and pass props properly, as a single map